From 6348708bfff46996e4786aefd21d07926152dc2d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?SimaMoto=2CRy=C5=8DTa?= <liangtai.s4@gmail.com>
Date: Wed, 27 Feb 2013 12:44:53 +0900
Subject: [PATCH] Detect openssl without using pkgconfig

* Allow to run on FreeBSD that don't have openssl.pc while it has
  libssl.so
* Detect libcompat.so to use ftime(3)
---
 configure.ac | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 3133b7f..4bcd973 100644
--- a/configure.ac
+++ b/configure.ac
@@ -53,8 +53,12 @@ AS_IF([test "x$enable_http" != "xno"], [
    AC_DEFINE([OP_ENABLE_HTTP], [1], [Enable HTTP support])
 ])
 AC_SUBST(openssl)
+AC_CHECK_LIB(ssl,SSL_read,
+	[AC_CHECK_HEADERS(openssl/ssl.h,LIBS="$LIBS -lssl -lcrypto",,)],,
+	-lcrypto -lssl)
 
-PKG_CHECK_MODULES([DEPS], [ogg >= 1.3 opus >= 1.0.1 ${openssl}])
+PKG_CHECK_MODULES([DEPS], [ogg >= 1.3 opus >= 1.0.1])
+AC_CHECK_LIB(compat,ftime)
 
 AC_ARG_ENABLE([fixed-point],
   AS_HELP_STRING([--enable-fixed-point], [Enable fixed-point calculation]),,
-- 
1.8.1.3

